test/src/materials/VarCouplingMaterial.C

```diff
@@ -20,12 +20,13 @@ VarCouplingMaterial::validParams()
   params.addParam<Real>("coef", 1.0, "The linear coefficient of the coupled var");
   params.addParam<bool>(
       "declare_old", false, "When True the old value for the material property is declared.");
+  params.addParam<TagName>("tag", "The solution vector to be coupled in");
   return params;
 }
 
 VarCouplingMaterial::VarCouplingMaterial(const InputParameters & parameters)
   : Material(parameters),
-    _var(coupledValue("var")),
+    _var(coupledVectorTagValue("var", "tag")),
```

Comment:
We should check if tag does exist or not. If it exists, we should call `coupledValue`. 

It does not sound right for me that we let `coupledVectorTagValue` involve  `coupledValue` when tag does not exist.

Reply:
If we have a tag name for the current solution vector, this will make sense. I can add that tag in `FEProblemBase` include `current_solution_tag`, `old_solutioner_tag`, `older_solution_tag` and associate them to the solution vectors. Then users can use them ad the default value of the `tag` parameter. What do you think?

Reply:
I have a better design now along this line. It is working but I will need more time for polishing.